index.ts

```diff
@@ -64,17 +64,32 @@ async function run(): Promise<void> {
     console.log('You can configure the deploy branch by setting the `deploy-branch` input for this action.')
 
     await exec.exec(`git init`, [], {cwd: `${workingDir}/public`})
-    await exec.exec(`git config user.name`, [github.context.actor], {
-      cwd: `${workingDir}/public`,
-    })
-    await exec.exec(`git config user.email`, [`${github.context.actor}@users.noreply.github.com`], {
+    await exec.exec(`git config user.name`, [core.getInput('git-config-name') || github.context.actor], {
       cwd: `${workingDir}/public`,
     })
+    await exec.exec(
+      `git config user.email`,
+      [core.getInput('git-config-email') || `${github.context.actor}@users.noreply.github.com`],
+      {
+        cwd: `${workingDir}/public`,
+      },
+    )
 
     await exec.exec(`git add`, ['.'], {cwd: `${workingDir}/public`})
-    await exec.exec(`git commit`, ['-m', `deployed via Gatsby Publish Action 🎩 for ${github.context.sha}`], {
-      cwd: `${workingDir}/public`,
-    })
+
+    const commitMessageInput = core.getInput('commit-message')
```

Comment:
This is great! Let's also include the default message here so we can remove the ternary operation below:

```suggestion
    const commitMessageInput = core.getInput('commit-message') || `deployed via Gatsby Publish Action 🎩 for ${github.context.sha}`
```